packages/web-app-files/tests/unit/components/SideBar/Links/PublicLinks/LinkEdit.spec.js

```diff
@@ -0,0 +1,613 @@
+            expect(linkCreateButton.exists()).toBeTruthy()
+            expect(linkCreateButton.attributes('disabled')).toBeFalsy()
+          })
+          it('should trigger "addLink" function if clicked', async () => {
+            const addLinkSpy = jest.spyOn(mapActions, 'addLink')
+            const wrapper = getMountedWrapper(
+              createStore({
+                linkInEdit: { name: 'Public Link', hasPassword: true },
+                publicLinkCapabilities: getLinkCapabilities({ enforcedExpireDate: true })
+              }),
+              { saving: false }
+            )
+            const linkCreateButton = wrapper.find(selectors.linkCreateButton)
+            expect(linkCreateButton.attributes('disabled')).toBeFalsy()
+
+            await linkCreateButton.trigger('click')
+            expect(wrapper.vm.saving).toBeTruthy()
+            expect(addLinkSpy).toHaveBeenCalledTimes(1)
+          })
+        })
+        describe('when existing link is being updated', () => {
+          it('should show link save button', () => {
+            const wrapper = getShallowMountedWrapper(
+              createStore({
+                linkInEdit: { id: 1224, name: 'Public Link' }
+              }),
+              { saving: false }
+            )
+
+            const linkSaveButton = wrapper.find(selectors.linkSaveButton)
+            expect(linkSaveButton.exists()).toBeTruthy()
+          })
+          it('should set link save button as disabled if form is not valid', () => {
+            const wrapper = getShallowMountedWrapper(
+              createStore({
+                linkInEdit: { id: 1224, name: 'Public Link' },
+                // invalid password field
+                // by default viewer role is selected with permission 1 (viewer)
+                // combined with enforced for password is enforce for link form
+                // since link password is enforced and password is not set, form will be invalid
+                publicLinkCapabilities: getLinkCapabilities({
+                  passwordEnforcedFor: {
+                    read_only: '1'
+                  }
+                })
+              }),
+              { saving: false }
+            )
+            const linkSaveButton = wrapper.find(selectors.linkSaveButton)
+
+            expect(linkSaveButton.attributes('disabled')).toBeTruthy()
+          })
+          it('should set link save button as disabled if form does not have any changes', () => {
+            const wrapper = getShallowMountedWrapper(
+              createStore({
+                linkInEdit: { id: 1224, name: 'Public Link', hasPassword: false, permissions: 1 },
+                publicLinkCapabilities: getLinkCapabilities({ enforcedExpireDate: true })
+              }),
+              { saving: false }
+            )
+            const linkSaveButton = wrapper.find(selectors.linkSaveButton)
+
+            expect(wrapper.vm.$_isValid).toBeTruthy()
+            expect(wrapper.vm.$_hasChanges).toBeFalsy()
```

Comment:
same argument here, I don't think we need to check the outcome of the functions, if the functions are refactored and the logic of how they are combined to set the `disabled` attribute but the attribute is set correctly the tests should not fail
Or if the functions are renamed, but the `disabled` attribute is set correctly, the tests should not fail.

Same applies to the next test

Reply:
fixed